### DISCUSSION TITLE
Do not use React.FC

### DISCUSSION THREAD

**Discussion by tkrotoff** [payloadcms/payload#7513] [Feature Requests & Ideas]
Payload source code, examples and documentation use React.FC (and/or friends: React.FunctionComponent, React.FunctionalComponent)

Instead of writing:

```TypeScript
const MyComponent: React.FC<Props> = ({ hello }) => {
  // ...
}
```

just write:

```TypeScript
function MyComponent({ hello }: Props) {
  // ...
}
```

React and TypeScript perfectly type this for years now.

There is a wide consensus that React.FC (and friends) is obsolete and “dangerous”:
- React.FC has been removed from create-react-app in January 2020: https://github.com/facebook/create-react-app/pull/8177
- Martin Hochel one of the core contributor of react.d.ts don't recommend it (2019): https://github.com/DefinitelyTyped/DefinitelyTyped/issues/34237#issuecomment-481247913
- React+TypeScript Cheatsheets also doesn't recommend it: https://github.com/typescript-cheatsheets/react#function-components
- Kent C Doods also in may 2021: https://kentcdodds.com/blog/how-to-write-a-react-component-in-typescript
- Backstage (an open source framework) has an ADR with the same conclusions since may 2020: https://backstage.io/docs/architecture-decisions/adrs-adr006/

Here a codemod to remove React.FC: https://github.com/gndelia/codemod-replace-react-fc-typescript

You can also add an ESLint rule to ban React.FC: https://stackoverflow.com/a/76818791

**Comment by JarrodMFlesch** [comment]
I agree with this change, looks like 600+ files would need to change. 370+ would be in packages/payload.

**Comment by jmikrut** [comment]
I'm into it too, would be a good PR for someone to make. Gonna convert to discussion and add this to the roadmap

**Comment by lahiruramesh** [comment]
I have made changes to this only for the tests folder(https://github.com/payloadcms/payload/pull/7528). This is my first contribution to the codebase. 

 remaining folders to remove React.RC
 docs
 examples
 packages
 templates
 
I used this command to refactor the code.

npx jscodeshift -- -t https://raw.githubusercontent.com/gndelia/codemod-replace-react-fc-typescript/main/dist/index.js --extensions=tsx --verbose=2 test\fields-relationship

**Comment by austingayler** [comment]
https://www.totaltypescript.com/you-can-stop-hating-react-fc

```
As of TypeScript 5.1 and React 18, React.FC is now officially 'fine'.

    It no longer implicitly includes children in the props type.
    It no longer breaks if you return undefined, string, or number.

I still recommend simply annotating props instead of using React.FC. But if you do use React.FC in your codebase, there's no reason to remove it.
```

**Comment by JarrodMFlesch** [comment]
I think if someone wanted to submit a PR with this change we would accept it. For now there are just other priorities, I think this would be great to have.
